packages/lambda-tiler/src/util/assets.provider.ts

```diff
@@ -0,0 +1,75 @@
+import { fsa } from '@chunkd/fs';
+import { LambdaHttpResponse, LambdaHttpRequest, HttpHeader } from '@linzjs/lambda';
+import { CachedConfig } from './config.cache.js';
+import { isGzip } from './cotar.serve.js';
+import { Etag } from './etag.js';
+import { NotFound, NotModified } from './response.js';
+import { CoSources } from './source.cache.js';
+
+export class AssetProvider {
+  /**
+   * Assets can be ready from the following locations.
+   *
+   * /home/blacha/config/build/assets # Local File
+   * /home/blacha/config/build/assets.tar.co # Local Cotar
+   * s3://linz-baesmaps/assets/ # Remote location
+   * s3://linz-basemaps/assets/assets-b4ff211a.tar.co # Remote Cotar
+   */
+
+  /** Path of the assets location */
+  path: string | undefined;
+
+  set(path?: string): void {
+    this.path = path;
+  }
+
+  async get(fileName: string): Promise<Buffer | null> {
+    if (this.path == null) return null;
+    // get assets file from cotar
+    if (this.path.endsWith('.tar.co')) return await this.getFromCotar(this.path, fileName);
+
+    // get assets file for directory
+    try {
+      const filePath = fsa.join(this.path, fileName);
+      return await fsa.read(filePath);
+    } catch (e: any) {
+      if (e.code === 404) return null;
+      throw e;
+    }
+  }
+
+  async getFromCotar(path: string, fileName: string): Promise<Buffer | null> {
+    const cotar = await CoSources.getCotar(path);
+    const data = await cotar.get(fileName);
+    return data ? Buffer.from(data) : data;
+  }
+
+  /**
+   *  Load a assets from local path or cotar returning the file back as a LambdaResponse
+   *
+   * This will also set two headers
+   * - Content-Encoding if the file starts with gzip magic
+   * - Content-Type from the parameter contentType
+   */
+  async serve(req: LambdaHttpRequest, file: string, contentType: string): Promise<LambdaHttpResponse> {
+    const config = req.query.get('config');
```

Comment:
config can not be handled here, as this would only affect the "serve" routes, this has to be handled at the top level route handler.